sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1517,14 +1517,13 @@ class Analyzer(override val catalogManager: CatalogManager)
         } else {
           a.copy(aggregateExpressions = buildExpandedProjectList(a.aggregateExpressions, a.child))
         }
-      // If the script transformation input contains Stars, expand it.
+      // When we construct ScriptTransformation, we pass UnresolvedStart as ScriptTransformation's
+      // input and handle query block like normal SELECT Clause as ScriptTransformation's child.
+      // In analyzer level, after child resolved, we just pass child's output as
+      // ScriptTransformation's input, then we can support using transformation with
+      // aggregation/having/window/lateral view and other features.
```

Comment:
I think we don't need this comment now (because this logic will be removed soon), so how about replacing it like this? 
```
// TODO: Remove this logic and see SPARK-34035
```

Reply:
> // TODO: Remove this logic and see SPARK-34035

Yea, done